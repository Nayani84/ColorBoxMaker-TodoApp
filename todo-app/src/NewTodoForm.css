.New-todo-form {
    display: flex;
    gap: 10px;
}

.New-todo-form input {
    flex: 1;
    padding: 5px;
    border: 1px solid #ccc;
    border-radius: 3px;
}

.New-todo-form button {
    padding: 5px 10px;
    background-color: #75de77;
    color: white;
    border: none;
    border-radius: 3px;
    cursor: pointer;
}

.New-todo-form button:hover {
    background-color: #069644f9;
}