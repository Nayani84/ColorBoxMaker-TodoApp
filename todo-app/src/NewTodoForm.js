import React, { useState } from "react";
import { v4 as uuid } from "uuid";
import './NewTodoForm.css';

const NewTodoForm = ({ addTodo }) => {
    const INITIAL_STATE = {
        task: ''
    }

    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleChange = (e) => {
        const { name, value } = e.target;
        setFormData(formData => ({
            ...formData,
            [name]: value
        }))
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        if (!formData.task.trim()) return;
        addTodo({ ...formData, id: uuid() });
        setFormData(INITIAL_STATE);
    }

    return (
        <form className="New-todo-form" onSubmit={handleSubmit}>
            <label htmlFor="task">Task :</label>
            <input
                id="task"
                type="text"
                name="task"
                placeholder="Task"
                value={formData.task}
                onChange={handleChange}
            />

            <button>Add Task</button>
        </form>
    );
}

export default NewTodoForm;


