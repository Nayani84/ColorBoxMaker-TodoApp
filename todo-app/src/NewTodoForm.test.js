import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

it("renders without crashing", function () {
    render(<NewTodoForm />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<NewTodoForm />);
    expect(asFragment()).toMatchSnapshot();
});

it("runs the create function on form submit", function () {
    const createMock = jest.fn();
    const { getByText, getByLabelText } = render(<NewTodoForm addTodo={createMock} />);

    const input = getByLabelText("Task :");
    fireEvent.change(input, { target: { value: "Test Task" } });

    const addBtn = getByText("Add Task");
    fireEvent.click(addBtn);
    expect(createMock).toHaveBeenCalled();
});
