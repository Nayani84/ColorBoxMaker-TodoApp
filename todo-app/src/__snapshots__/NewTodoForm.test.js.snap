// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="New-todo-form"
  >
    <label
      for="task"
    >
      Task :
    </label>
    <input
      id="task"
      name="task"
      placeholder="Task"
      type="text"
      value=""
    />
    <button>
      Add Task
    </button>
  </form>
</DocumentFragment>
`;
